dtfs-central-api/src/utils/report-period.ts

```diff
@@ -27,3 +29,45 @@ export const getPreviousReportPeriodStart = ({ month, year }: MonthAndYear): Mon
 
 export const isEqualReportPeriodStart = (reportPeriodStart1: MonthAndYear, reportPeriodStart2: MonthAndYear): boolean =>
   reportPeriodStart1.year === reportPeriodStart2.year && reportPeriodStart1.month === reportPeriodStart2.month;
+
+export const getCurrentReportPeriodForBankSchedule = (bankReportPeriodSchedule: ReportPeriodSchedule[]): ReportPeriod => {
```

Comment:
The logic in this function is pretty complicated. I've tried to name the variables such that it is easier to figure out what is going on, but I'm keen to have any suggestions on how to make the `reportPeriodInPreviousYear` logic easier to digest

Reply:
`reportPeriodInPreviousYear` implies this could be when both start and end are in the previous year (e.g. 10-12), but I don't think that's what this is used for. I think this is the case when the schedule overlaps the year boundary?

I'm not sure about using `currentYear` as that implies today's year, but this could be last year.

I think we can simplify this a bit by first finding the current relevant schedule, the returning the result based on that. I'm not sure about the naming, but it would be something like this:
```
export const getCurrentReportPeriodForBankSchedule = (bankReportPeriodSchedule: ReportPeriodSchedule[]): ReportPeriod => {
  const previousMonthDate = subMonths(new Date(), 1);
  const targetMonth = getOneIndexedMonth(previousMonthDate);
  const targetYear = previousMonthDate.getFullYear();

  const currentSchedule = bankReportPeriodSchedule.find((schedule) => {
    const isYearOverlapping = schedule.startMonth > schedule.endMonth && (targetMonth >= schedule.startMonth || targetMonth <= schedule.endMonth);
    const isNonYearOverlapping = schedule.startMonth <= targetMonth && targetMonth <= schedule.endMonth;
    return isYearOverlapping || isNonYearOverlapping;
  });

  if (!currentSchedule) {
    throw new Error(`Failed to get current report period start for bank with report schedule: ${currentSchedule}`);
  }

  const yearOverlapping = currentSchedule.startMonth > currentSchedule.endMonth;

  return {
    start: {
      month: currentSchedule.startMonth,
      year: yearOverlapping ? targetYear - 1 : targetYear,
    },
    end: {
      month: currentSchedule.endMonth,
      year: targetYear,
    },
  };
};
```

Reply:
As discussed, I've changed this implementation as we also need to worry about whether or not the report period goes into the future. I'll resolve this conversation, but note that it needs to be looked at again :)